libssh2: add mbedtls to PKG_BUILD_DEPENDS

micmac1 opened the pull request:
This adds to PKG_BUILD_DEPENDS to prevent built failures. A comment is
left in case anybody stumbles over this in the future.

Closes #12108

Signed-off-by: Sebastian Kemper <sebastian_ml@gmx.net>

Maintainer: @jslachta 
Compile tested: ath79 master
Run tested: No change to package

Description:
Hi all,

With this change mbedtls gets pulled in when needed (fixes build failures like described in the referenced issue). In case the user selects the symbol LIBSSH2_OPENSSL mbedtls doesn't get pulled in, because it's not needed.

I did not bump the revision because package remains unchanged.

Kind regards,
Seb

micmac1 commented:
Had to fix spelling in comment and commit. "built" vs "build" was driving me mad %) I hope spelling is correct now.

micmac1 commented:
And another misspelling removed :)

neheb commented:
@neheb is more in favor of removing openssl support.

jslachta commented:
The ability to choose crypto dependency is causing me headaches.

I would like to drop openssl support.

micmac1 commented:
>@neheb is more in favor of removing openssl support.

We had that for a (short) while, until people wanted it back :-)

https://github.com/openwrt/packages/issues/5550

neheb commented:
Why doesn't libcurl hit this problem?

micmac1 commented:
Maybe if done the other way around (drop mbetls) nobody would notice/complain. Sounds funny, but might be true. So far I can only remember people asking for openssl.

neheb commented:
Good point. None of the dependencies depend on mbedtls. Some depend on OpenSSL.

micmac1 commented:
> Why doesn't libcurl hit this problem?

You mean the problem nmap has? Because curl uses
"+LIBCURL_SSH2:libssh2". If you mean openssl vs. mbedtls (vs. wolfssl
vs. gnutls), I'd say because curl can also be built without SSL support
(--without-wolfssl, --without-gnutls, --without-ssl, --without-mbedtls).
libssh2 needs a TLS lib on the other hand.

micmac1 commented:
Good morning,

I'm sure you saw the idea from @cotequeiroz in the other issue. I like this better. So I changed this pull request accordingly (PKG_BUILD_DEPENDS not needed anymore).

I also switched the default to OpenSSL, to try it on for size. This way mbedtls is still available as an option, but as @neheb said yesterday, all packages that depend on libssh2 _and_ need their own TLS lib depend on OpenSSL.

What do you think about this?

neheb commented on libs/libssh2/Makefile:
```diff
@@ -53,10 +66,10 @@ CMAKE_OPTIONS += \
+ifeq ($(CONFIG_LIBSSH2_MBEDTLS),y)
 	CMAKE_OPTIONS += -DCRYPTO_BACKEND=mbedTLS
+else
+	CMAKE_OPTIONS += -DCRYPTO_BACKEND=OpenSSL
```
I assume this can be simplified to

```
-DCRYPTO_BACKEND=$(if $(CONFIG_LIBSSH2_MBEDTLS),mbedTLS,OpenSSL)
```

and placed with the main group.

micmac1 commented:
> >  	CMAKE_OPTIONS += -DCRYPTO_BACKEND=mbedTLS
> +else
> +	CMAKE_OPTIONS += -DCRYPTO_BACKEND=OpenSSL
>
> I assume this can be simplified to
>
> ```
> -DCRYPTO_BACKEND=$(if $(CONFIG_LIBSSH2_MBEDTLS),mbedTLS,OpenSSL)
> ```
>
> and placed with the main group.

Sure, just pushed it.

jslachta commented:
Merged, thank you!
